csrc/fusion_segmenter.cpp

```diff
@@ -510,7 +530,7 @@ void SegmentedFusion::deserialize(const serde::SegmentedFusion* buffer) {
       buffer->num_vals(),
       " values.");
   NVF_ERROR(
-      complete_fusion_->exprs().size() <= buffer->num_exprs(),
+      complete_fusion_->unordered_exprs().size() <= buffer->num_exprs(),
```

Comment:
Is this a necessary change? I thought even after the clone, we would still only serialize expressions leading to outputs, but not necessarily everything in the IR.

Reply:
In this case, we're not serializing the `complete_fusion_` but the mapping from `complete_fusion_` to segmented group. The deserialized `complete_fusion_` should match the original up to the expected number of expressions.

`IrContainer::deterministic_exprs_map` should match `IrContainer::unordered_exprs` rather than `Fusion::exprs`.

Reply:
`Fusion::exprs` doesn't contain all expressions necessary for the fusion. We need `SegmentCandidateFinder::resolveScalarsInGroup` to find expressions related to the fusion's scalars.